src/mbgl/text/glyph_set.cpp

```diff
@@ -87,83 +86,113 @@ void justifyLine(std::vector<PositionedGlyph> &positionedGlyphs, const std::map<
         }
     }
 }
+  
+float GlyphSet::determineIdeographicLineWidth(const std::u16string& logicalInput, const float spacing, float maxWidth) const
+{
+    float totalWidth = 0;
+    
+    // totalWidth doesn't include the last character for magical tuning reasons. This makes the algorithm a little
+    //  more agressive about trying to fit the text into fewer lines, taking advantage of the tolerance for going a little
+    //  over maxWidth
+    for (uint32_t i = 0; i < logicalInput.size() - 1; i++) {
```

Comment:
Could be, but then we have to add extra logic to remove the last character, and I don't think the code gets much simpler. Or am I missing something?